### PR TITLE
spec updates

### PR DESCRIPTION
**Pull request by arnetheduck**
* make several constants uint64 to help minimize casting
* document data type woes - will have to revisit these
* change comment style on fields and constants to make room for better
comments
* add BLSVerify and BLSAddPbkeys facades to insulate spec code from
milagro
* fix proof of possession type
* drop explicitly ordered container fields from ssz - there's an issue
open to sort this out before committing it to the spec

### CODE REVIEW

File: beacon_chain/spec/datatypes.nim [hunk 1]
```diff
@@ -27,86 +27,224 @@ import
...
+  MIN_BALANCE* = 2'u64^4 ##\
+  ## Minimum balance in ETH before a validator is removed from the validator
+  ## pool
+
+  MAX_BALANCE_CHURN_QUOTIENT* = 2^5 ##\
+  ## At most `1/MAX_BALANCE_CHURN_QUOTIENT` of the validators can change during
+  ## each validator registry change.
+
+  GWEI_PER_ETH* = 10'u64^9 # Gwei/ETH
+  BEACON_CHAIN_SHARD_NUMBER* = not 0'u64
```
Comment (mratsim) [inline beacon_chain/spec/datatypes.nim]: We should add a comment to say that this is 2^64 - 1

File: beacon_chain/spec/datatypes.nim [hunk 2]
```diff
@@ -200,6 +334,12 @@ type
     Activation = 0
     Exit = 1
 
+  SignatureDomain* {.pure.} = enum
```
Comment (mratsim) [inline beacon_chain/spec/datatypes.nim]: The {.pure.} pragma is just placebo now.

File: beacon_chain/state_transition.nim [hunk 3]
```diff
@@ -156,7 +107,7 @@ func processRandaoReveal(state: var BeaconState,
 func processPoWReceiptRoot(state: var BeaconState, blck: BeaconBlock): bool =
   for x in state.candidate_pow_receipt_roots.mitems():
     if blck.candidate_pow_receipt_root == x.candidate_pow_receipt_root:
-      x.votes.inc
+      x.votes += 1
```
Comment (mratsim) [inline beacon_chain/state_transition.nim]: Should we had something to the style guide with regards to inc/dec vs `+=`/`-=`? (Actually should we start a style guide?)

File: beacon_chain/spec/datatypes.nim (hunk 2 above)
Comment (arnetheduck) [reply beacon_chain/spec/datatypes.nim]: oh is it? I actually looked it up in the manual where it says it still does something.. sigh

File: beacon_chain/state_transition.nim (hunk 3 above)
Comment (arnetheduck) [reply beacon_chain/state_transition.nim]: they're (unfortunately) not the same - `inc` unconditionally takes an `int` as parameter while `+=` takes a `T` / same type.. the latter seems to make more sense in general. what's funny is that functionally they're the same as long as you assume two's complement and similar bit-sizes, but that's stretching your luck

File: beacon_chain/spec/datatypes.nim (hunk 1 above)
Comment (arnetheduck) [reply beacon_chain/spec/datatypes.nim]: pushed

**Pull request merged**